pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -588,10 +553,32 @@ static int NpyTypeToJSONType(PyObject *obj, JSONTypeContext *tc, int npyType,
             PRINTMARK();
             return JT_NULL;
         }
-        GET_TC(tc)->longValue = (JSINT64)longVal;
-        GET_TC(tc)->PyTypeToJSON = NpyDatetime64ToJSON;
-        return ((PyObjectEncoder *)tc->encoder)->datetimeIso ? JT_UTF8
-                                                             : JT_LONG;
```

Comment:
im not so familiar with this syntax.  is the ?:; here equivalent to the if/else you've changed this to?

There is still `GET_TC(tc)->PyTypeToJSON = NpyDatetime64ToJSON` on 559.  You're not getting rid of this one?

Reply:
That is the ternary operator, so basically `<EXPRESSION> ? <VALUE_IF_TRUE> : <VALUE_IF_FALSE>`

This is used elsewhere in the module; just copying over

Reply:
i think i get it, thanks